source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -735,79 +779,53 @@ namespace
         {
             if (!filter || filter(elem))
             {
-                readLookGroup(elem, rootPrim, stage, mapper);
+                readLookGroup(elem, stage, mapper);
             }
         }
 
         // Create additional connections
-        makeCollectionIncludeConnections(doc->getCollections(), rootPrim, mapper);
-        makeLookInheritConnections(doc->getLooks(), rootPrim, mapper);
+        makeCollectionIncludeConnections(doc->getCollections(), parent, mapper);
+        makeLookInheritConnections(doc->getLooks(), parent, mapper);
     }
 
-    void validateNodesHaveNodedefs(DocumentPtr doc)
+    void readDocument(const DocumentPtr& doc, PvtStage* stage, const RtReadOptions* options)
     {
-        for (auto elem : doc->getChildren())
+        // Set the source location.
+        const string& uri = doc->getSourceUri();
+        if (!uri.empty())
         {
-            if (elem->isA<Node>())
-            {
-                NodePtr node = elem->asA<Node>();
-                const RtToken nodedefName = resolveNodeDefName(node);
-                if (nodedefName == EMPTY_TOKEN)
-                {
-                    throw ExceptionRuntimeError("No matching nodedef was found for node '" + node->getName() + "'");
-                }
-            }
+            stage->addSourceUri(uri);
```

Comment:
Think we want to keep separate uris for definitions vs content. This allows us to search relative to content documents first then definition files. We have this separation for saving w/o including dependent files in other places.